test/e2e/nsx_ipblocksinfo_test.go

```diff
@@ -0,0 +1,168 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	nsxutil "github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+var (
+	ipBlocksInfoCRDName = "ip-blocks-info"
+	defaultOrg          = "default"
+	defaultProject      = "project-quality"
+	defaultVPCProfile   = "default"
+)
+
+func TestIPBlocksInfo(t *testing.T) {
+	t.Run("case=InitialIPBlocksInfo", InitialIPBlocksInfo)
+	t.Run("case=CustomIPBlocksInfo", CustomIPBlocksInfo)
+}
+
+func InitialIPBlocksInfo(t *testing.T) {
+	privateTGWIPCIDRs, externalIPCIDRs := getDefaultIPBlocksCidrs(t)
+	assertIPBlocksInfo(t, privateTGWIPCIDRs, externalIPCIDRs)
+}
+
+func CustomIPBlocksInfo(t *testing.T) {
+	// Create Private IPBlocks
+	ipBlockName := "ipblocksinfo-test-10.0.0.0-netmask-28"
+	ipBlockCidr := "10.0.0.0/28"
+	err := testData.nsxClient.IPBlockClient.Patch(defaultOrg, defaultProject, ipBlockName, model.IpAddressBlock{
+		Cidr:       &ipBlockCidr,
+		Visibility: common.String("PRIVATE"),
+	})
+	require.NoError(t, err)
+	defer func() {
+		testData.nsxClient.IPBlockClient.Delete(defaultOrg, defaultProject, ipBlockName)
```

Comment:
for those defer func, could you add checking the result and log error if error is not nil?

Reply:
Added, thanks!